### PR TITLE
Add the ability to get serialized configuration of a task

### PR DESCRIPTION
**Pull request by IgnatBeresnev**
A proposal for an intermediate solution for #2873, which should help with debugging configuration and compatibility issues.

cc @aSemy

### CODE REVIEW

File: runners/gradle-plugin/api/gradle-plugin.api [hunk 1]
```diff
@@ -173,6 +173,10 @@ public final class org/jetbrains/dokka/gradle/SourceSetKotlinGistConfigurationKt
 	public static final fun configureWithKotlinSourceSet (Lorg/jetbrains/dokka/gradle/GradleDokkaSourceSetBuilder;Lorg/jetbrains/kotlin/gradle/plugin/KotlinSourceSet;)V
 }
 
+public final class org/jetbrains/dokka/gradle/internal/AbstractDokkaTaskExtensionsKt {
+	public static final fun buildJsonConfiguration (Lorg/jetbrains/dokka/gradle/AbstractDokkaTask;)Ljava/lang/String;
+}
+
```
Comment (aSemy) [inline runners/gradle-plugin/api/gradle-plugin.api:179]: Shouldn't this be excluded from the API dump because it's marked as `InternalDokkaApi`?

**Comment by aSemy** [comment]
Ahh clever solution, I like it.

I have a couple of observations.

(btw you linked the wrong task in the description, it should be #2873, not #2936)

File: runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/internal/AbstractDokkaTaskExtensions.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+ * Serializes [DokkaConfiguration] of this [AbstractDokkaTask] as pretty json
+ *
+ * Should be used for debugging only, no guarantees are given for the support of this API.
+ *
+ * Better alternative should be introduced as part of [#2873](https://github.com/Kotlin/dokka/issues/2873).
+ */
+@InternalDokkaApi
+fun AbstractDokkaTask.buildJsonConfiguration(): String {
+    return this.buildDokkaConfiguration().toPrettyJsonString()
+}
```
Comment (aSemy) [inline runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/internal/AbstractDokkaTaskExtensions.kt:18]: Did you consider adding a parameter, `pretty: Boolean`, and then conditionally returning `toPrettyJsonString()` or `toCompactJsonString()`? 

It's fine as it is, and I don't see a strong argument for returning compact JSON. I'm just curious.

Comment (IgnatBeresnev) [reply runners/gradle-plugin/src/main/kotlin/org/jetbrains/dokka/gradle/internal/AbstractDokkaTaskExtensions.kt:18]: Good point, definitely won't hurt 👍 will add it

**Comment by IgnatBeresnev** [comment]
@aSemy could I ask you to review the PR since you've already had a look? :) I've implemented the pretty print suggestion

**Review by aSemy — COMMENTED** [review]
I approve this PR, but GitHub won't let me tick the 'Approve' button 

image

**Comment by IgnatBeresnev** [comment]
I was actually wondering about that..

Thanks!

**Pull request merged**